onnx/defs/tensor/utils.cc

```diff
@@ -167,4 +167,83 @@ void resizeShapeInference_opset7_to_10(InferenceContext& ctx) {
   }
 }
 
+void splitShapeInference(InferenceContext& ctx, bool allow_negative_axis) {
+  for (int i = 0; i < static_cast<int>(ctx.getNumOutputs()); ++i) {
+    propagateElemTypeFromInputToOutput(ctx, 0, i);
+  }
+  if (!hasNInputShapes(ctx, 1)) {
+    return;
+  }
+
+  const auto& shape = ctx.getInputType(0)->tensor_type().shape();
+  int rank = shape.dim_size();
+  int axis = static_cast<int>(getAttribute(ctx, "axis", 0));
+  if (axis < (allow_negative_axis ? -rank : 0) || axis >= rank) {
+    fail_type_inference(
+        "Invalid value of attribute 'axis'. Rank=",
+        rank,
+        " Value=",
+        axis);
+  }
+  if (allow_negative_axis && axis < 0) {
```

Comment:
no need checking allow_negative_axis here.

Reply:
Done, sorry for the large latency!